### PR TITLE
T375566 : Wikiwrapped pt 1

### PR DESCRIPTION
**Pull request by l-olson1214**
**Phabricator:** 
https://phabricator.wikimedia.org/T375566

### Notes
* To have entry point, test in build 170 OR uncomment lines 178-199 in WMFProfileViewModel - this PR has the entry point commented out

### Test Steps
1. Open app (read note above)
2. Navigate to profile
3. Navigate to Year in Review
4. Ensure it shows, you can navigate through the slides, press the done button, or the next button

### Screenshots/Videos

https://github.com/user-attachments/assets/ce7dcd4e-8cd0-4240-9c2d-1b92f55d1875

Simulator Screenshot - iPhone 15 Pro - 2024-10-01 at 12 59 25
Simulator Screenshot - iPhone 15 Pro - 2024-10-01 at 15 50 09
Simulator Screenshot - iPhone 15 Pro - 2024-10-01 at 15 50 24
Simulator Screenshot - iPhone 15 Pro - 2024-10-01 at 15 50 29

### CODE REVIEW

File: WMFComponents/Sources/WMFComponents/Components/Year in Review/WMFSlideShow.swift [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+                    VStack(alignment: .leading, spacing: 16) {
+                        Image(slides[slide].imageName, bundle: .module)
+                            .frame(maxWidth: .infinity, alignment: .center)
+                        Text(slides[slide].title)
+                            .font(Font(WMFFont.for(.boldTitle1)))
+                            .foregroundStyle(Color(uiColor: theme.text))
+                        Text(slides[slide].subtitle)
+                            .font(Font(WMFFont.for(.title3)))
+                            .foregroundStyle(Color(uiColor: theme.text))
+                        Spacer()
```
Comment (mazevedofs) [inline WMFComponents/Sources/WMFComponents/Components/Year in Review/WMFSlideShow.swift:34]: We might need to wrap this up in a scroll view for larger font sizes so the text isn't cropped.

**Review by mazevedofs — CHANGES_REQUESTED** [review]
Looks good! I found some small things and made a couple of optional comments that are not at all needed.

File: WMFComponents/Sources/WMFComponents/Coordinator/ProfileCoordinatorDelegate.swift [hunk 2]
```diff
@@ -16,4 +16,5 @@ public enum ProfileAction {
     case login
     case logout
     case logDonateTap
+    case yearInReviewTap
```
[the remark this answers is not in the document]
Comment (l-olson1214) [reply WMFComponents/Sources/WMFComponents/Coordinator/ProfileCoordinatorDelegate.swift:19]: Gotcha! Thanks :D

File: WMFComponents/Sources/WMFComponents/Components/Profile/WMFProfileViewModel.swift [hunk 3]
```diff
@@ -156,8 +160,20 @@ enum ProfileState {
...
+//                            hasNotifications: false,
+//                            isDonate: false,
+//                            isLoadingDonateConfigs: false,
+//                            action: {
+//                                coordinatorDelegate?.handleProfileAction(.showYearInReview)
+//                            }
+//                        )
                     ],
-                    subtext: nil
+                    subtext: nil //subtext: "Log in or create an account to get an improved year in review next year"
```
Comment (mazevedofs) [inline WMFComponents/Sources/WMFComponents/Components/Profile/WMFProfileViewModel.swift:176]: You can delete this comment here, I believe this text would belong to the logged out version

File: WMFComponents/Sources/WMFComponents/Components/Year in Review/WMFSlideShow.swift [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+public struct WMFSlideShow: View {
+    @ObservedObject var appEnvironment = WMFAppEnvironment.current
+    @Binding private var currentSlide: Int
+    
+    var theme: WMFTheme {
+        return appEnvironment.theme
+    }
+    
+    let slides: [any SlideShowProtocol]
```
Comment (mazevedofs) [inline WMFComponents/Sources/WMFComponents/Components/Year in Review/WMFSlideShow.swift:11]: You don't need the `any` keyword here or in line 13 with this type of protocol

**Review by mazevedofs — CHANGES_REQUESTED** [review]

**Review by mazevedofs — APPROVED** [review]
Looking good!

**Pull request merged**